### PR TITLE
Bump base and template-haskell bounds to enable GHC 9.10

### PR DESCRIPTION
**Pull request by jhrcek**
Hello, 
this package is one of the transitive dependencies of haskell-language-server, which I'm trying to enable for GHC 9.10. Could you please release new 9.10-supporting version of this package to hackage?

Apart from bumping base and template-haskell upper bounds, I'm also I replacing defunct travis CI config with haskell-ci -generated github actions config.

You can see the new CI config in action in my fork: https://github.com/jhrcek/constraints-extras/pull/1

It seems that the build is failing with ghc 8.0.2 and 8.2.2. with error like this:
```
Building executable 'readme' for constraints-extras-0.4.0.0..
[1 of 1] Compiling Main             ( README.lhs, /__w/constraints-extras/constraints-extras/dist-newstyle/build/x86_64-linux/ghc-8.2.2/constraints-extras-0.4.0.0/x/readme/build/readme/readme-tmp/Main.o )

README.lhs:47:3: error:
    • Expected kind ‘* -> *’, but ‘Fam ()’ has kind ‘*’
    • In the data instance declaration for ‘Fam’
   |
47 | > data instance Fam () :: * -> * where
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
Error: cabal-3.10.2.0: Failed to build exe:readme from constraints-extras-0.4.0.0.
```

Would you prefer me trying to fix it, or would you be ok with dropping these 2 oldest GHC versions from tested-with / CI?

Also would you be ok with replacing the existing CI config with a one generated by haskell-ci?
Although the generated one is more verbose, I find it more comprehensive and easier to maintain (it bumps actions versions etc.)

### CODE REVIEW

**Comment by jhrcek** [comment]
Hello @ali-abrar (sorry for at-mentioning, but you seem to be the most active contributor),
could you please take a look and comment on this PR?

**Comment by ali-abrar** [comment]
Thanks. We can drop support for the older GHCs as long as we note it in the changelog and update the tested-with field in the cabal file. I'd prefer to keep 8.6.5 if it isn't too much trouble (appears to be the last official release of ghcjs) and 8.10 (the most recent unofficial ghcjs release).

**Comment by ali-abrar** [comment]
The new GitHub action file looks fine to me. Thank you for the contribution.

File: ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Revision history for constraints-extras
 
+## Unreleased
```
Comment (ali-abrar) [inline ChangeLog.md:3]: Should this be a revision in hackage or a new version?

**Review by ali-abrar — APPROVED** [review]
Just need to set the version (or decide this is a revision) so that these changes can be released. They can, however, be merged.

**Comment by ali-abrar** [comment]
> Just need to set the version (or decide this is a revision) so that these changes can be released. They can, however, be merged. 

Once they pass CI, that is.

**Comment by ali-abrar** [comment]
https://github.com/obsidiansystems/constraints-extras/releases/tag/v0.4.0.1

**Comment by jhrcek** [comment]
Thank you :heart:

**Pull request merged**